[YAML-CLI2] ModuleNotFoundError: No module named 'flask'

### Background / motivation

(In yaml-cli2 branch.)

Calina and I each independently encountered this error, after making trivial changes to the code.

Here's the one I saw: (from [this log](https://github.com/oceanprotocol/pdr-backend/actions/runs/7471934534/job/20333111056)):
```text
==================================== ERRORS ====================================
____________ ERROR collecting pdr_backend/accuracy/test/test_app.py ____________
ImportError while importing test module '/home/runner/work/pdr-backend/pdr-backend/pdr_backend/accuracy/test/test_app.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
pdr_backend/accuracy/test/test_app.py:7: in <module>
    from pdr_backend.accuracy.app import (
pdr_backend/accuracy/app.py:7: in <module>
    from flask import Flask, jsonify
E   ModuleNotFoundError: No module named 'flask'
=========================== short test summary info ============================
ERROR pdr_backend/accuracy/test/test_app.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 6.[33](https://github.com/oceanprotocol/pdr-backend/actions/runs/7471934534/job/20333111056#step:8:34)s ===============================
```

### Discussion

pdr_backend/accuracy/app.py has been around for a while. It imports flask.

What's new is that there's now a unit test for it, at accuracy/test/test_app.py. This unit test imports the app, which imports flask.

Somehow that test got merged into yaml-cli2 branch without CI passing :(. Because adding that test would have caused an obvious CI failure. cc @kdetry @idiom-bytes 

Regardless, now, CI is failing.

### TODO

Ideal: add flask to dependencies. This is preferred, if flask isn't too heavy and the test isn't too slow.

Possible backups:
- make the unit test faster
- make the unit test less often (eg weekly)

### On the final result

PR & commit are linked below.

Flask _was_ light enough (whew), and unit test does run fast (whew).

--- a/setup.py
+++ b/setup.py
@@ -15,6 +15,7 @@ install_requirements = [
     "enforce_typing",
     "eth-account",
     "eth-keys",
+    "flask",
     "matplotlib",
     "mypy",
     "numpy",

